lib/Request.js

```diff
@@ -53,8 +53,18 @@ function Request(method, url, params, opts, tries, callback) {
             return;
         }
 
-        document = libxml.parseHtml(document,
-                                    { baseUrl: location.href, huge: true });
+        var parse_mode = opts.parse_as;
+        if (parse_mode === 'auto')
+            parse_mode = getResponseType(res.headers['content-type']);
+        try {
+            if (parse_mode === 'xml')
+                document = libxml.parseXml(document, {baseUrl: location.href, huge: true});
```

Comment:
Hmm, was it always throwing on parse errors or is this new with `parseXml`?

Can you try adding the [XML parse option](https://github.com/libxmljs/libxmljs/wiki/Libxmljs) `errors: false` so we can avoid the try/catch?

Reply:
Libxmljs docs says that unrecoverable errors are "thrown and need to be caught". I don't know is there any fatal errors in html mode, though. But anyway using try-catch even for html is safer, just in case. Why don't you like it?

Xml throws even with `errors: false`. Looks like this option is only about recoverable errors.